commit 943b37624c4836755687c936b5214543b031cfdc
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Tue May 8 19:34:02 2018 +0200

    Revert "Drop Go 1.8 compatibility"
    
    This reverts commit 22441a088d5630ddd2e971eae68074d2b645f1b7.

diff --git a/cluster.go b/cluster.go
index 85d2616..5c161d9 100644
--- a/cluster.go
+++ b/cluster.go
@@ -178,7 +178,10 @@ func setTimeouts(dependencies []*dependencies) {
 
 // Set leader notification channels on all servers.
 func instrumentConfigs(t testing.TB, logger *log.Logger, dependencies []*dependencies) *election.Tracker {
-	t.Helper()
+	helper, ok := t.(testingHelper)
+	if ok {
+		helper.Helper()
+	}
 
 	tracker := election.NewTracker(logger)
 
@@ -251,7 +254,10 @@ func connectLoobackTransports(dependencies []*dependencies) {
 // Bootstrap the cluster, including in the initial configuration of each voting
 // server.
 func bootstrapCluster(t testing.TB, logger *log.Logger, dependencies []*dependencies) {
-	t.Helper()
+	helper, ok := t.(testingHelper)
+	if ok {
+		helper.Helper()
+	}
 
 	// Figure out which servers should be part of the initial
 	// configuration.
@@ -300,3 +306,8 @@ func bootstrapCluster(t testing.TB, logger *log.Logger, dependencies []*dependen
 func newRaft(d *dependencies) (*raft.Raft, error) {
 	return raft.NewRaft(d.Conf, d.FSM, d.Logs, d.Stable, d.Snaps, d.Trans)
 }
+
+// For compatibility with Go <1.9
+type testingHelper interface {
+	Helper()
+}
diff --git a/control.go b/control.go
index 9f97d79..944a404 100644
--- a/control.go
+++ b/control.go
@@ -488,7 +488,10 @@ func WaitLeader(t testing.TB, raft *raft.Raft, timeout time.Duration) {
 }
 
 func waitLeader(ctx context.Context, t testing.TB, raft *raft.Raft) {
-	t.Helper()
+	helper, ok := t.(testingHelper)
+	if ok {
+		helper.Helper()
+	}
 
 	check := func() bool {
 		return raft.Leader() != ""
@@ -499,7 +502,10 @@ func waitLeader(ctx context.Context, t testing.TB, raft *raft.Raft) {
 // Poll the given function at the given internval, until it returns true, or
 // the given context expires.
 func wait(ctx context.Context, t testing.TB, f func() bool, interval time.Duration, message string) {
-	t.Helper()
+	helper, ok := t.(testingHelper)
+	if ok {
+		helper.Helper()
+	}
 
 	start := time.Now()
 	for {
commit a89446071ba8b86866147fe0961fe60ed653c134
Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Tue May 8 20:57:55 2018 +0200

    Restore Go 1.8 compatibility

diff --git a/control.go b/control.go
index 944a404..27ccd99 100644
--- a/control.go
+++ b/control.go
@@ -77,7 +77,10 @@ func (c *Control) Close() {
 // When calling this method there must be no leader in the cluster and server
 // transports must all be disconnected from eacher.
 func (c *Control) Elect(id raft.ServerID) *Term {
-	c.t.Helper()
+	helper, ok := c.t.(testingHelper)
+	if ok {
+		helper.Helper()
+	}
 
 	c.logger.Printf("[DEBUG] raft-test: elect: start (server %s)", id)
 
